File: packages/sdk/sdk-php/tests/organisations_test.php
<?php
declare(strict_types=1);

require_once __DIR__ . "/../src/index.php";

use Phaseo\Sdk\Phaseo;

function assert_true(bool $condition, string $message): void
{
    if (!$condition) {
        throw new RuntimeException($message);
    }
}

final class FakeOrganisationsClient extends \Phaseo\Gen\Client
{
    public array $calls = [];

    public function __construct()
    {
        parent::__construct("https://example.test");
    }

    public function request(string $method, string $path, ?array $query = null, ?array $headers = null, $body = null)
    {
        $this->calls[] = [$method, $path, $query, $headers, $body];
        return [
            "ok" => true,
            "limit" => 2,
            "offset" => 3,
            "total" => 1,
            "organisations" => [
                [
                    "organisation_id" => "org_123",
                    "name" => "Anthropic",
                    "country_code" => "US",
                    "colour" => "#D97706",
                ],
            ],
        ];
    }
}

$client = new Phaseo(
    apiKey: "test",
    basePath: "https://api.phaseo.app/v1",
    enableDeprecationWarnings: false
);

$fake = new FakeOrganisationsClient();
$ref = new ReflectionClass($client);
$prop = $ref->getProperty("client");
$prop->setAccessible(true);
$prop->setValue($client, $fake);

$response = $client->listOrganisations(["limit" => "2", "offset" => "3"]);
assert_true(($response["ok"] ?? null) === true, "expected ok payload");
assert_true(($response["organisations"][0]["organisation_id"] ?? null) === "org_123", "expected organisation id");
assert_true($fake->calls === [
    ["GET", "/organisations", ["limit" => "2", "offset" => "3"], null, null],
], "expected request call for organisations discovery");

echo "php organisations tests ok" . PHP_EOL;

File: packages/sdk/sdk-ruby/tests/organisations_test.rb
require "minitest/autorun"
require_relative "../lib/index"

class OrganisationsTest < Minitest::Test
  def test_list_organisations_returns_payload
    client = PhaseoSdk::Phaseo.new(
      api_key: "test",
      enable_deprecation_warnings: false
    )

    calls = []
    client.raw_client.define_singleton_method(:request) do |method:, path:, query: nil, headers: nil, body: nil|
      calls << [method, path, query, headers, body]
      {
        "ok" => true,
        "limit" => 2,
        "offset" => 3,
        "total" => 1,
        "organisations" => [
          {
            "organisation_id" => "org_123",
            "name" => "Anthropic",
            "country_code" => "US",
            "colour" => "#D97706"
          }
        ]
      }
    end

    response = client.list_organisations("limit" => "2", "offset" => "3")

    assert_equal true, response["ok"]
    assert_equal "org_123", response["organisations"][0]["organisation_id"]
    assert_equal [["GET", "/organisations", { "limit" => "2", "offset" => "3" }, nil, nil]], calls
  end
end
